enzy_htp/structure/structure_region/api.py

```diff
@@ -174,13 +192,19 @@ class is only topology related. If create an obj for
     
     @property
     def involved_residues(self) -> List[Residue]:
-        """get all involved residues in the region"""
-        result = set()
+        """Get all involved residues in the region"""
+        result:List[Residue] = list()
+        unique = set()
         for atom in self.atoms:
             res = atom.parent
-            if not isinstance(res, ResidueCap):
-                result.add(res)
-        return list(result)
+            if isinstance(res, ResidueCap):
+                continue
+
+            if res not in unique:
+                unique.add(res)
+                result.append(res)
```

Comment:
`.add()` and `not in` is duplicated in terms of making element unique.

Reply:
Or you are making the order not the same as how atom appears?

Reply:
I want to maintain the order of the `Residue()`'s. Since `not in` is slow-ish with a `list()`, I use the faster `set()` to check if the `Residue` has been added.

Reply:
Sounds good